app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala

```diff
@@ -0,0 +1,119 @@
+import uk.gov.hmrc.viewmodels.Text.Literal
+import uk.gov.hmrc.viewmodels.{Checkboxes, NunjucksSupport}
+
+import java.time.LocalDate
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController @Inject()(
+    override val messagesApi: MessagesApi,
+    sessionRepository: SessionRepository,
+    navigator: Navigator,
+    identify: IdentifierAction,
+    getData: DataRetrievalAction,
+    requireData: DataRequiredAction,
+    formProvider: SelectAnyTaxpayersThisEnterpriseIsAssociatedWithFormProvider,
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      // retrieve list of taxpayers TODO get the actual list from page
+      import SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.taxpayers
+
+      val preparedForm = request.userAnswers.get(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      val items: Seq[Checkboxes.Checkbox] = taxpayers.map { taxpayer =>
+        Checkboxes.Checkbox(label = Literal(taxpayer.nameAsString), value = s"${taxpayer.taxpayerId}")
+      }
+      val checkboxes = Checkboxes.set(field = preparedForm("value"), items = items)
+
+      val json = Json.obj(
+        "form" -> preparedForm,
+        "mode"       -> mode,
+        "checkboxes" -> checkboxes
+      )
+
+      renderer.render("enterprises/selectAnyTaxpayersThisEnterpriseIsAssociatedWith.njk", json).map(Ok(_))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        formWithErrors => {
+
+          val json = Json.obj(
+            "form"       -> formWithErrors,
+            "mode"       -> mode
+          )
+
+          renderer.render("enterprises/selectAnyTaxpayersThisEnterpriseIsAssociatedWith.njk", json).map(BadRequest(_))
+        },
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage, value))
+            _              <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage, mode, updatedAnswers))
+      )
+  }
+
+  def toCheckboxes(taxpayers: Set[JsObject]): Seq[Checkboxes.Checkbox] = taxpayers.map { json =>
+    val name = json.validate[Taxpayer].getOrElse(throw new IllegalArgumentException("")).nameAsString
+    Checkboxes.Checkbox(label = Literal(name), value = json.toString())
+  }.toSeq
```

Comment:
Not used anywhere. It's similar to line 65 in the onPageLoad method above.

Reply:
Removed